### PR TITLE
Add case of virsh domifaddr

### PR DESCRIPTION
**Pull request by chloerh**
- VIRT-99481: [domifaddr] get vm's ip address by domifaddr

Signed-off-by: Haijiao Zhao <haizhao@redhat.com>

### CODE REVIEW

**Comment by chloerh** [comment]
Test result:

 (1/4) type_specific.io-github-autotest-libvirt.virsh.domifaddr.positive_test.guest_agent_started: PASS (53.58 s)
 (2/4) type_specific.io-github-autotest-libvirt.virsh.domifaddr.negative_test.vm_off: PASS (8.02 s)
 (3/4) type_specific.io-github-autotest-libvirt.virsh.domifaddr.negative_test.vm_not_exist: PASS (7.97 s)
 (4/4) type_specific.io-github-autotest-libvirt.virsh.domifaddr.negative_test.guest_agent_stopped: PASS (51.77 s)

**Review by nanli1 — COMMENTED** [review]
Just confirm ,For the step 3 and step 4 , should they be executed in order or whatever

**Comment by chloerh** [comment]
> Just confirm ,For the step 3 and step 4 , should they be executed in order or whatever

@nanli1 Thanks for your comment :handshake:  Confirmed with feature owner, step3 & 4 could be executed separately.

**Review by nanli1 — APPROVED** [review]

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_domifaddr.cfg [hunk 1]
```diff
@@ -0,0 +1,24 @@
+- virsh.domifaddr:
+    type = virsh_domifaddr
+    start_vm = no
+    variants:
+        - positive_test:
+            status_error = 'no'
+            variants:
+                - guest_agent_started:
+                    start_vm = 'yes'
+                    pre_cmd_in_vm = 'systemctl start qemu-guest-agent'
```
Comment (Yingshun) [inline libvirt/tests/cfg/virsh_cmd/domain/virsh_domifaddr.cfg:10]: There are some functions about guest agent in libvirt_vm.py, I think we can reuse vm.prepare_guest_agent() or  vm.set_state_guest_agent()

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_domifaddr.cfg [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            status_error = 'yes'
+            variants:
+                - vm_off:
+                    error_msg = 'Requested operation is not valid: domain is not running'
+                - vm_not_exist:
+                    test_vm_name = 'non-existent'
+                    error_msg = 'failed to get domain'
+                - guest_agent_stopped:
+                    start_vm = 'yes'
+                    pre_cmd_in_vm = 'systemctl stop qemu-guest-agent'
```
Comment (Yingshun) [inline libvirt/tests/cfg/virsh_cmd/domain/virsh_domifaddr.cfg:22]: Same as above.

**Review by Yingshun — COMMENTED** [review]
Others LGTM

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_domifaddr.cfg (hunk 2 above)
Comment (chloerh) [reply libvirt/tests/cfg/virsh_cmd/domain/virsh_domifaddr.cfg]: Updated test result:

```
 (1/4) type_specific.local.virsh.domifaddr.positive_test.guest_agent_started: PASS (18.06 s)
 (2/4) type_specific.local.virsh.domifaddr.negative_test.vm_off: PASS (2.47 s)
 (3/4) type_specific.local.virsh.domifaddr.negative_test.vm_not_exist: PASS (2.43 s)
 (4/4) type_specific.local.virsh.domifaddr.negative_test.guest_agent_stopped: PASS (20.56 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2022-11-21T17.09-6b3669d/results.html
JOB TIME   : 44.00 s
```

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_domifaddr.cfg (hunk 1 above)
Comment (chloerh) [reply libvirt/tests/cfg/virsh_cmd/domain/virsh_domifaddr.cfg]: Updated. Thanks :+1:

**Review by Yingshun — APPROVED** [review]

**Pull request merged**